### PR TITLE
[reward] doc.go

### PR DESCRIPTION
**Pull request by laggu**
## Proposed changes

- update doc.go of reward package

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kjhman21** [comment]
Please checkout the result by executing `godoc -http=:6060  -goroot=$HOME/go`. We need more new lines.

File: reward/doc.go [hunk 1]
```diff
@@ -14,5 +14,62 @@
...
 
-// Package reward implements the Klaytn reward system.
+/*
+Reward Package
+Package Reward implements the Klaytn Reward System.
+Klaytn Reward System manages stakingInfo and distributes block reward.
+
+Managing stakingInfo
+Klaytn uses weightedRandom policy to choose a block proposer.
+It means, the percentage of becoming a block proposer depends on how much klay a node has staked.
```
Comment (KimKyungup) [inline reward/doc.go]: Why don't you describe more detail where staked klay is?

File: reward/doc.go [hunk 2]
```diff
@@ -14,5 +14,62 @@
...
+Therefore, a node staked more than others will have more opportunities than other nodes.
+
+StakingInfo is a data including stakingAmount and addresses(node, staking, reward, poc, kir).
```
Comment (KimKyungup) [inline reward/doc.go]: `addresses(node` -> `addresses (node`
`poc, kir` -> `poc, and kir`

File: reward/doc.go [hunk 3]
```diff
@@ -14,5 +14,62 @@
...
+stakingAmount is balance how much each node has staked.
```
Comment (KimKyungup) [inline reward/doc.go]: `balance` -> `a balance`

File: reward/doc.go [hunk 4]
```diff
@@ -14,5 +14,62 @@
...
+addresses are included stakingAddress which store stakingAmount and reward, poc, kir which get a block reward when a block has been created.
+StakingInfo is made every 86400 blocks(stakingInterval) and used in a next interval.
```
Comment (KimKyungup) [inline reward/doc.go]: `blocks(stakingInterval)` -> `blocks (stakingInterval)`

File: reward/doc.go [hunk 5]
```diff
@@ -14,5 +14,62 @@
...
+
+Distributing Reward
+Klaytn distributes the reward of a block to proposer, PoC and KIR.
+
+The detail information of PoC and KIR is available on Klaytn docs.
+ PoC - https://docs.klaytn.com/klaytn/token_economy#proof-of-contribution
+ KIR - https://docs.klaytn.com/klaytn/token_economy#klaytn-improvement-reserve
+
+Configures related to reward system such as mintingAmount, ratio and unitPrice are determined by Klaytn governance.
+Configures are all saved as rewardConfig every epoch block(default 604800 blocks) and managed by rewardConfigCache
```
Comment (KimKyungup) [inline reward/doc.go]: `Configures are all` -> `All configures are` ??
Please mark the period at the end of line.

File: reward/doc.go [hunk 6]
```diff
@@ -14,5 +14,62 @@
...
+
+A proposer which has made a current block will get the reward of the block.
+A block reward is calculated by following steps.
+ 1. calculate totalReward by adding mintingAmount and totalTxFee(unitPrice * gasUsed).
+ 2. divide totalReward by ratio(default 34/54/12).
+ 3. distribute reward to each address(proposer, PoC, KIR).
+
```
Comment (KimKyungup) [inline reward/doc.go]: Please put a space before the parenthesis.

File: reward/doc.go (hunk 1 above)
Comment (jeonghonah) [reply reward/doc.go]: klay --> KLAY

Comment (jeonghonah) [inline reward/doc.go]: klay --> KLAY

```suggestion
It means, the percentage of becoming a block proposer depends on how much KLAY a node has staked.
```

File: reward/doc.go (hunk 2 above)
Comment (jeonghonah) [inline reward/doc.go]: stakingAmount --> staking amount

Comment (jeonghonah) [inline reward/doc.go]: (node, staking, reward, poc, kir) --> (node, staking, reward, poc and kir)

File: reward/doc.go [hunk 7]
```diff
@@ -14,5 +14,62 @@
...
+Klaytn Reward System manages stakingInfo and distributes block reward.
+
+Managing stakingInfo
+Klaytn uses weightedRandom policy to choose a block proposer.
+It means, the percentage of becoming a block proposer depends on how much klay a node has staked.
+Therefore, a node staked more than others will have more opportunities than other nodes.
+
+StakingInfo is a data including stakingAmount and addresses(node, staking, reward, poc, kir).
+stakingAmount is balance how much each node has staked.
+addresses are included stakingAddress which store stakingAmount and reward, poc, kir which get a block reward when a block has been created.
```
Comment (jeonghonah) [inline reward/doc.go]: I'm not sure but how about this?:

`are included` --> `includes`?

File: reward/doc.go [hunk 8]
```diff
@@ -14,5 +14,62 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
 
-// Package reward implements the Klaytn reward system.
+/*
+Reward Package
+Package Reward implements the Klaytn Reward System.
+Klaytn Reward System manages stakingInfo and distributes block reward.
```
Comment (jeonghonah) [inline reward/doc.go]: Klaytn Reward System manages --> The Klaytn Reward System manages

File: reward/doc.go (hunk 1 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 2 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 3 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 4 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 5 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 6 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 1 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 2 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 7 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go (hunk 8 above)
Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go [hunk 9]
```diff
@@ -14,5 +14,71 @@
...
+Klaytn uses weightedRandom policy to choose a block proposer.
+It means, the percentage of becoming a block proposer depends on how much KLAY a node has staked.
+Therefore, a node staked more than others will have more opportunities than other nodes.
+
+StakingInfo is a data including stakingAmount and addresses (node, staking, reward, PoC and KIR).
+StakingAmount is a balance how much KLAY each node has staked. Only CCO can stake KLAY.
+Each CCO stakes KLAY by a smart contract called staking contract.
+StakingAmount is derived by checking balance of staking contract.
+A StakingInfo has 5 types of addresses. All addresses are obtained by the addressBookContract which is pre-deployed in the genesis block.
+stakingAddress is addresses of staking contracts of CCO. reward, poc and kir address are the addresses which get a block reward when a block has been created.
```
Comment (jeonghonah) [inline reward/doc.go]:
```suggestion
stakingAddress are addresses of staking contracts of CCO. Reward, PoC and KIR addresses are the addresses which get a block reward when a block has been created.
```

File: reward/doc.go [hunk 10]
```diff
@@ -14,5 +14,71 @@
...
+KIR - https://docs.klaytn.com/klaytn/token_economy#klaytn-improvement-reserve
+
+Configures related to the reward system such as mintingAmount, ratio and unitPrice are determined by the Klaytn governance.
+All configures are saved as rewardConfig on every epoch block (default 604800 blocks) and managed by a rewardConfigCache.
+
+A proposer which has made a current block will get the reward of the block.
+A block reward is calculated by following steps.
+First, calculate totalReward by adding mintingAmount and totalTxFee (unitPrice * gasUsed).
+Second, divide totalReward by ratio (default 34/54/12).
+Last, distribute reward to each address(proposer, PoC, KIR).
```
Comment (jeonghonah) [inline reward/doc.go]:
```suggestion
Last, distribute reward to each address (proposer, PoC, KIR).
```

File: reward/doc.go [hunk 11]
```diff
@@ -14,5 +14,71 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
 
-// Package reward implements the Klaytn reward system.
+/*
+Package Reward
+
+Package Reward implements the Klaytn Reward System.
+The Klaytn Reward System manages stakingInfo and distributes block reward.
```
Comment (jeonghonah) [inline reward/doc.go]:
```suggestion
The Klaytn Reward System manages stakingInfo and distributes block rewards.
```

File: reward/doc.go [hunk 12]
```diff
@@ -14,5 +14,71 @@
...
+StakingAmount is derived by checking balance of staking contract.
+A StakingInfo has 5 types of addresses. All addresses are obtained by the addressBookContract which is pre-deployed in the genesis block.
+stakingAddress is addresses of staking contracts of CCO. reward, poc and kir address are the addresses which get a block reward when a block has been created.
+StakingInfo is made every 86400 blocks (stakingInterval) and used in a next interval.
+
+	type StakingInfo struct {
+		BlockNum              uint64
+		CouncilNodeAddrs      []common.Address // Address of Council
+		CouncilStakingAddrs   []common.Address // Address of Staking contract which holds staking balance
+		CouncilRewardAddrs    []common.Address // Address of Council account which will get block reward
```
Comment (jeonghonah) [inline reward/doc.go]:
```suggestion
		CouncilRewardAddrs    []common.Address // Address of Council account which will get a block reward
```

File: reward/doc.go [hunk 13]
```diff
@@ -14,5 +14,69 @@
...
 // along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
 
-// Package reward implements the Klaytn reward system.
+/*
+Package Reward implements the Klaytn Reward System.
+The Klaytn Reward System manages stakingInfo and distributes block rewards.
+
+Managing stakingInfo
+
+Klaytn uses weightedRandom policy to choose a block proposer.
```
Comment (jeonghonah) [inline reward/doc.go]:
```suggestion
Klaytn uses WeightedRandom policy to choose a block proposer.
```

Comment (laggu) [reply reward/doc.go]: reflect

File: reward/doc.go [hunk 14]
```diff
@@ -14,5 +14,69 @@
...
+
+Klaytn uses WeightedRandom policy to choose a block proposer.
+It means, the percentage of becoming a block proposer depends on how much KLAY a node has staked.
+Therefore, a node staked more than others will have more opportunities than other nodes.
+
+StakingInfo is a data including stakingAmount and addresses (node, staking, reward, PoC and KIR).
+StakingAmount is a balance how much KLAY each node has staked. Only CCO can stake KLAY.
+Each CCO stakes KLAY by a smart contract called staking contract.
+StakingAmount is derived by checking balance of staking contract.
+A StakingInfo has 5 types of addresses. All addresses are obtained by the addressBookContract which is pre-deployed in the genesis block.
```
Comment (jeonghonah) [inline reward/doc.go]: A StakingInfo --> StakingInfo

Comment (laggu) [reply reward/doc.go]: reflected

File: reward/doc.go [hunk 15]
```diff
@@ -14,5 +14,69 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
 
-// Package reward implements the Klaytn reward system.
+/*
+Package Reward implements the Klaytn Reward System.
```
Comment (ehnuje) [inline reward/doc.go]: Reward -> reward ?

File: reward/doc.go [hunk 16]
```diff
@@ -14,5 +14,69 @@
...
+The Klaytn Reward System manages stakingInfo and distributes block rewards.
```
Comment (ehnuje) [inline reward/doc.go]: Reward System -> reward system?

File: reward/doc.go [hunk 17]
```diff
@@ -14,5 +14,69 @@
...
+
+Managing stakingInfo
+
+Klaytn uses WeightedRandom policy to choose a block proposer.
+It means, the percentage of becoming a block proposer depends on how much KLAY a node has staked.
+Therefore, a node staked more than others will have more opportunities than other nodes.
```
Comment (ehnuje) [inline reward/doc.go]: a node staked more than others -> a node with stakes more than others?

File: reward/doc.go [hunk 18]
```diff
@@ -14,5 +14,69 @@
...
+Distributing Reward
+
+Klaytn distributes the reward of a block to proposer, PoC and KIR.
+The detail information of PoC and KIR is available on Klaytn docs.
+
+PoC - https://docs.klaytn.com/klaytn/token_economy#proof-of-contribution
+
+KIR - https://docs.klaytn.com/klaytn/token_economy#klaytn-improvement-reserve
+
+Configures related to the reward system such as mintingAmount, ratio and unitPrice are determined by the Klaytn governance.
```
Comment (ehnuje) [inline reward/doc.go]: Configures -> Configurations

File: reward/doc.go [hunk 19]
```diff
@@ -14,5 +14,69 @@
...
+All configures are saved as rewardConfig on every epoch block (default 604800 blocks) and managed by a rewardConfigCache.
```
Comment (ehnuje) [inline reward/doc.go]: Configures -> Configurations

Comment (ehnuje) [reply reward/doc.go]: Adding a comma like 604,800 would be helpful for readers.

File: reward/doc.go [hunk 20]
```diff
@@ -14,5 +14,69 @@
...
+
+A proposer which has made a current block will get the reward of the block.
+A block reward is calculated by following steps.
+First, calculate totalReward by adding mintingAmount and totalTxFee (unitPrice * gasUsed).
+Second, divide totalReward by ratio (default 34/54/12).
```
Comment (ehnuje) [inline reward/doc.go]: 34%=proposer, 54%=PoC, 12%=KIR?

**Pull request merged**